### DISCUSSION TITLE
Translations with Headless Documents

### DISCUSSION THREAD

**Discussion by neusta-diaspero** [pimcore/pimcore#12021] [Q&A / Support] [answered]
Hello Pimcore-Community,

regarding **Pimcore Headless Documents** I'm just wondering how to use **translations** of headless documents and access them via GraphQL. Any hints on that?

Kind Regards,
Claudio.

**Comment by fashxp** [comment]
Hi,
what kind of translations you mean? 
BR

**Reply by neusta-diaspero** [reply]
Hello Christian,

yes, let me try to explain this more in detail.

I'm wondering, if there is a way of fetching the linked translations (itself headless documents too) of a headless document via GraphQL Query. The "master" headless document could be e.g. the English variant referring to multiple translations like German …

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/36693399/166474102-dad99055-132e-46d5-9757-950b7b272098.png">

For data objects it is clear: By querying the desired language of an object via `defaultLanguage: "…"` attribute translated properties are provides. Any idea or best practice on doing so for headless documents?

Kind Regards,
Claudio.

**Comment by dvesh3** [comment, marked as answer]
Hi @neusta-diaspero, Unfortunately, it is not possible to fetch the linked translations from Headless Document in Datahub query (not even supported for Standard Documents), so for now you have to query the translated documents directly.

I have created an issue to add support in data hub bundle https://github.com/pimcore/data-hub/issues/528

**Reply by neusta-diaspero** [reply]
Thank you. Unfortunately, this does not really help. It is difficult to load translated documents whose IDs the consumer does not know.  I am really looking forward this feature.
